rtos/rtos_idle.c

```diff
@@ -22,14 +22,50 @@
 
 #include "rtos/rtos_idle.h"
 #include "platform/mbed_sleep.h"
+#include "mbed_critical.h"
+#include "hal/ticker_api.h"
+#include "hal/lp_ticker_api.h"
+#include "cmsis_os2.h"
+#include <limits.h>
+
+static ticker_event_t idle_loop_event;
+
+void idle_loop_handler(uint32_t id)
+{
+    (void)id;
+}
 
 static void default_idle_hook(void)
 {
-    /* Sleep: ideally, we should put the chip to sleep.
-       Unfortunately, this usually requires disconnecting the interface chip (debugger).
-       This can be done, but it would break the local file system.
-    */
+#if DEVICE_LOWPOWERTIMER
```

Comment:
We got test for that - lp ticker in HAL tests - sleep and wake up by lp ticker. if that fails, then this would fail also. lp ticker had always a requirement to wake up from sleep and also deepsleep. This one does at the moment only sleep, if it does not work it's a bug.